pkg/splunk/controller/configmap_test.go

```diff
@@ -41,3 +42,62 @@ func TestApplyConfigMap(t *testing.T) {
 	}
 	spltest.ReconcileTester(t, "TestApplyConfigMap", &current, revised, createCalls, updateCalls, reconcile, false)
 }
+
+func TestGetConfigMap(t *testing.T) {
+	current := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "defaults",
+			Namespace: "test",
+		},
+	}
+
+	client := spltest.NewMockClient()
+	namespacedName := types.NamespacedName{Namespace: current.GetNamespace(), Name: current.GetName()}
+
+	_, err := GetConfigMap(client, namespacedName)
+	if err == nil {
```

Comment:
Nit: We could expect an error **"NotFound"** as follows:

```
if err.Error() != "NotFound" {
    t.Errorf("Failed to recognize non-existence of configMap") 
}
```
Also we could make **NotFound** as an entry in **pkg/splunk/common/messages.go**

Reply:
That is good to have, but, it has its own drawbacks if at all we want to cover for all the error conditions. For example, a returned error string can be variable in length, values, in which case, we need to do string comparisons like suffix, prefix, contains etc..., which will drastically reduce the performance. For every returned error, we are anyhow printing the error, so we should be good IMO, and all we need is a good descriptive comment above each test case.

Added to that, looking at the coverage report, we can make sure that there are no holes(to the maximum extent possible).

Reply:
Performance, functionality and code coverage wise a comment describing the expected error sounds good @sgontla.